kubernetes/data_source_kubernetes_endpoints.go

```diff
@@ -0,0 +1,35 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func dataSourceKubernetesEndpoint() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceKubernetesEndpointRead,
+		Schema: map[string]*schema.Schema{
+			"metadata": namespacedMetadataSchema("endpoints", true),
+			"subset": {
+				Type:        schema.TypeSet,
+				Description: "Set of addresses and ports that comprise a service. More info: https://kubernetes.io/docs/concepts/services-networking/service/#services-without-selectors",
+				Optional:    true,
+				Elem:        schemaEndpointsSubset(),
+				Set:         hashEndpointsSubset(),
+			},
+		},
+	}
+}
+
+func dataSourceKubernetesEndpointRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Please add a V1 suffix to the function name and make it plural to align with `kind` and `apiVersion` -- `dataSourceKubernetesEndpointsV1Read`.